test/audience.robot

```diff
@@ -27,17 +30,42 @@ Token with correct audience is accepted
     Should Contain   ${out}   404
 
 Token with correct audience in multiple option is accepted
+    [Tags]   audience
     ${uuid}   Generate UUID
     ${se_config}   Get SE config
     ${token}   Get token   scope=-s openid   opts=--aud="${se_config.endpoint} ${uuid}"
     ${url}   SE URL   audience-test-${uuid}
     ${rc}   ${out}   Curl Error   ${url}
     Should Contain   ${out}   404
 
-Token with invalid multiple audiences is rejected
+Token with invalid multiple audiences is rejected (permissive)
+    [Tags]   audience   permissive
+    ${out}   Get token with invalid multiple audience
+    ${ret}   Should Match Regexp   ${out}   40[134]
+
+Token with invalid multiple audiences is rejected (strict)
+    [Tags]   audience   strict
+    ${out}   Get token with invalid multiple audience
+    ${ret}   Should Match Regexp   ${out}   401
+
+
+*** Keywords ***
+
+Get token with random audience
+    ${token}   Get token
+    ${uuid}   Generate UUID
+    ${url}   SE URL   audience-test-${uuid}
+    ${token}   Get token   scope=-s openid   opts=--aud=${uuid}
+    ${rc}   ${out}   Curl Error   ${url}
+    [Return]   ${out}
+
+Get token with invalid multiple audience
+    ${token}   Get token
     ${uuid}   Generate UUID
+    ${url}   SE URL   audience-test-${uuid}
+    ${rc}   ${out}   Curl Put Success   /etc/services  ${url}
```

Comment:
Why do we do a `PUT` first? Is it so that in permissive mode 404 is not a valid status code? we should not accept it then in permissive mode.
Shouldn't we do the same also in the keyword `Get token with random audience`?

Reply:
Yes, my intention was to avoid checks on 404 status code but this is not correctly implemented in the test. And you are right, we should use the same logic for both tests.